php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1498 @@
+      .is(f.variableDeclaration(
+        b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER),
+        b.optional(f.newTuple96(b.token(EQU), STATIC_SCALAR()))
+      ));
+  }
+
+  public NamespaceNameTree NAMESPACE_NAME() {
+    return b.<NamespaceNameTree>nonterminal(PHPLexicalGrammar.NAMESPACE_NAME)
+      .is(b.firstOf(
+          f.namespaceName(
+            b.optional(b.token(PHPPunctuator.NS_SEPARATOR)),
+            b.zeroOrMore(f.newTuple4(
+              b.firstOf(b.token(PHPLexicalGrammar.IDENTIFIER)),
+              b.token(PHPPunctuator.NS_SEPARATOR)
+            )),
+            b.token(PHPLexicalGrammar.IDENTIFIER)),
+          f.namespaceName(
+            b.token(PHPKeyword.NAMESPACE),
+            b.token(PHPPunctuator.NS_SEPARATOR),
+            b.zeroOrMore(f.newTuple6(
+              b.firstOf(b.token(PHPLexicalGrammar.IDENTIFIER)),
+              b.token(PHPPunctuator.NS_SEPARATOR)
+            )),
+            b.token(PHPLexicalGrammar.IDENTIFIER)))
+      );
+  }
+
+  public UseClauseTree USE_CLAUSE() {
+    return b.<UseClauseTree>nonterminal(PHPLexicalGrammar.USE_CLAUSE).is(
+      f.useClause(
+        NAMESPACE_NAME(),
+        b.optional(
+          f.newTuple91(
+            b.token(PHPKeyword.AS),
+            b.token(PHPLexicalGrammar.IDENTIFIER)))));
+  }
+
+  public ClassDeclarationTree CLASS_DECLARATION() {
+    return b.<ClassDeclarationTree>nonterminal(PHPLexicalGrammar.CLASS_DECLARATION)
+      .is(f.classDeclaration(
+        b.optional(b.firstOf(b.token(ABSTRACT), b.token(FINAL))),
+        b.token(CLASS),
+        IDENTIFIER(),
+        b.optional(f.newTuple50(b.token(EXTENDS), NAMESPACE_NAME())),
+        b.optional(f.newTuple30(b.token(IMPLEMENTS), INTERFACE_LIST())),
+        b.token(LCURLYBRACE),
+        b.zeroOrMore(CLASS_MEMBER()),
+        b.token(RCURLYBRACE)
+      ));
+  }
+
+  public ClassDeclarationTree TRAIT_DECLARATION() {
+    return b.<ClassDeclarationTree>nonterminal(PHPLexicalGrammar.TRAIT_DECLARATION)
+      .is(f.traitDeclaration(
+        b.token(TRAIT),
+        IDENTIFIER(),
+        b.token(LCURLYBRACE),
+        b.zeroOrMore(CLASS_MEMBER()),
+        b.token(RCURLYBRACE)
+      ));
+  }
+
+  public ClassDeclarationTree INTERFACE_DECLARATION() {
+    return b.<ClassDeclarationTree>nonterminal(PHPLexicalGrammar.INTERFACE_DECLARATION)
+      .is(f.interfaceDeclaration(
+        b.token(INTERFACE),
+        IDENTIFIER(),
+        b.optional(f.newTuple26(b.token(EXTENDS), INTERFACE_LIST())),
+        b.token(LCURLYBRACE),
+        b.zeroOrMore(CLASS_MEMBER()),
+        b.token(RCURLYBRACE)
+      ));
+  }
+
+
+  public ClassMemberTree CLASS_MEMBER() {
+    return b.<ClassMemberTree>nonterminal(PHPLexicalGrammar.CLASS_MEMBER)
+      .is(b.firstOf(
+        METHOD_DECLARATION(),
+        CLASS_VARIABLE_DECLARATION(),
+        CLASS_CONSTANT_DECLARATION(),
+        TRAIT_USE_STATEMENT()
+      ));
+  }
+
+  public ClassPropertyDeclarationTree CLASS_CONSTANT_DECLARATION() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)